### ISSUE TITLE
Problems of stdin, stdout in mintty

### ISSUE THREAD

**Issue by chentian-4477** [mintty/mintty#1159]
Hello, everyone.
I found a problem about the wrong sequence of the "printf" when using gcc compiler in Msys2.
There are no prints when I call the "printf" function in the "menu" block.
In Msys2, using command " ./a.exe ", there are no prints at first, when I input the digits, but the prints appears after I quit the program.
In Window10, double click the a.exe, it prints the message, prints again when I input the digits again. It alway print the messages until I quit the program. (correct running)
In Debian11.3, complie with the same code, run the result "./a.out", running is correct too, just like in Windows10.

The c codes are from "C Primer Plus (6th Edition)"-chapter 9:
Listing 9.9 usehotel.c
Listing 9.10 hotel.c
Listing 9.11 hotel.h

Running in Msys2(wrong running):
![image](https://user-images.githubusercontent.com/64013077/171107151-314694d2-ff23-4289-9e53-c46619f9e76a.png)

Running in Debian11 3(correct running):
![image](https://user-images.githubusercontent.com/64013077/171107275-eb88bd20-c71d-4f1e-a210-c2229b8c7f1f.png)

Running in Windows10(correct running):
![image](https://user-images.githubusercontent.com/64013077/171107315-2678ddca-eda6-486f-a04d-67d16f4d41ae.png)

Something wrong with Msys2 gcc? How to fix it?

[hotel.zip](https://github.com/mintty/mintty/files/8802592/hotel.zip)

Another issue opened:  https://github.com/msys2/MINGW-packages/issues/11624
**Someone told me that was the problems of 'stdin' 'stdout' in mintty.**

**Comment by k-takata** [comment]
You are using the MINGW64 shell. So, when you use gcc, a normal 64-bit program will be built.
If you run a normal 64-bit program in mintty, it won't work interactively unless you enable ConPTY. (ConPTY is disabled by default in MSYS2.)
Another way is to use the MSYS shell to build your program. Then, your program will be built as an msys2 program. You should be able to run it inside mintty as expected, but it may not work outside mintty.

**Comment by mintty** [comment]
Run `mintty -o ConPTY=true` to verify this is not a terminal issue.
Set MSYS=enable_pcon before starting a session to enable ConPTY in it.
See also https://github.com/mintty/mintty/wiki/Tips#inputoutput-interaction-with-alien-programs

**Comment by chentian-4477** [comment]
> 
Hello, @mintty ,
I am so sorry that I just see your answer. And your answer is active when I run "mintty -o ConPTY=true".
And could you please help me to set the "MSYS=enable_pcon" ? I am so weak about setting the mintty. 
I have tried the "/etc/fstab" and "/mingw64.ini", unfortunately they are failed.

**Comment by mintty** [comment]
You can set `ConPTY=true` in a mintty config file (see manual for locations).
Or you can set environment variable MSYS manually in cmd.exe before starting mintty (less appreciated), 
or globally in your Windows system settings, section environment variables which is quite well hidden nowadays...

**Comment by chentian-4477** [comment]
> You can set `ConPTY=true` in a mintty config file (see manual for locations). Or you can set environment variable MSYS manually in cmd.exe before starting mintty (less appreciated), or globally in your Windows system settings, section environment variables which is quite well hidden nowadays...

Thank you so much! I have set 'ConPTY=true' in ' .minttyrc' in my home folder. It worked. Now I can use the mintty in my MSYS2. It provides me a almost real linux terminal environment in my windows 10.

Thanks again. You have been a great help to me.
